public/app/features/transformers/editors/CalculateFieldTransformerEditor.tsx

```diff
@@ -436,44 +407,43 @@ export class CalculateFieldTransformerEditor extends React.PureComponent<
             allowMultiple={false}
             className="width-18"
             stats={[options.reducer]}
-            onChange={this.onStatsChange}
+            onChange={onStatsChange}
             defaultStat={ReducerID.sum}
           />
         </InlineField>
       </>
     );
-  }
+  };
 
   //---------------------------------------------------------
   // Cumulative Operator
   //---------------------------------------------------------
 
-  onCumulativeStatsChange = (stats: string[]) => {
+  const onCumulativeStatsChange = (stats: string[]) => {
     const reducer = stats.length ? (stats[0] as ReducerID) : ReducerID.sum;
 
-    const { reduce } = this.props.options;
-    this.updateCumulativeOptions({ ...reduce, reducer });
+    const { cumulative } = options;
+    updateCumulativeOptions({ ...cumulative, reducer });
```

Comment:
This isn't a 1:1 change.
I discovered this bug in testing. When updating the reducer, the field would reset since it's  spreading the wrong variable.